pkg/swap/geth.go

```diff
@@ -0,0 +1,222 @@
+	if err != nil {
+		return "", fmt.Errorf("get accounts: %w", err)
+	}
+
+	if !contains(ethAccounts, g.ethAccount) {
+		return "", fmt.Errorf("eth account %s not found", g.ethAccount)
+	}
+
+	req := ethRequest{
+		ID:      "0",
+		JsonRPC: "1.0",
+		Method:  "eth_sendTransaction",
+		Params: []ethRequestParams{{
+			From:  g.ethAccount,
+			To:    to,
+			Value: addPrefix("0x", fmt.Sprintf("%x", float64ToBigInt(amount))),
+			Gas:   addPrefix("0x", fmt.Sprintf("%x", EthGasPrice)),
+		}},
+	}
+
+	resp := new(struct {
+		ID      string `json:"id"`
+		JsonRPC string `json:"jsonrpc"`
+		Result  string `json:"result"`
+	})
+
+	if err = requestJSON(ctx, g.httpClient, http.MethodPost, "/", req, &resp); err != nil {
+		return "", err
+	}
+
+	return resp.Result, nil
+}
+
+// sendBZZ makes BZZ token deposit
+func (g *GethClient) SendBZZ(ctx context.Context, to string, amount float64) (tx string, err error) {
+	ethAccounts, err := g.ethAccounts(ctx)
+	if err != nil {
+		return "", fmt.Errorf("get accounts: %w", err)
+	}
+
+	if !contains(ethAccounts, g.ethAccount) {
+		return "", fmt.Errorf("eth account %s not found", g.ethAccount)
+	}
+
+	req := ethRequest{
+		ID:      "0",
+		JsonRPC: "1.0",
+		Method:  "eth_sendTransaction",
+		Params: []ethRequestParams{{
+			From: g.ethAccount,
+			To:   g.bzzTokenAddress,
+			Data: "0x40c10f19" + fmt.Sprintf("%064s", removePrefix("0x", to)) + fmt.Sprintf("%064x", float64ToBigInt(amount)),
+			Gas:  addPrefix("0x", fmt.Sprintf("%x", BzzGasPrice)),
+		}},
+	}
+
+	resp := new(struct {
+		ID      string `json:"id"`
+		JsonRPC string `json:"jsonrpc"`
+		Result  string `json:"result"`
+	})
+
+	if err = requestJSON(ctx, g.httpClient, http.MethodPost, "/", req, &resp); err != nil {
+		return "", err
+	}
+
+	return resp.Result, nil
+}
+
+// ethAccounts returns list of accounts
+func (g *GethClient) ethAccounts(ctx context.Context) (a []string, err error) {
+	req := ethRequest{
+		ID:      "0",
+		JsonRPC: "1.0",
+		Method:  "eth_accounts",
+		Params:  []ethRequestParams{},
+	}
+
+	resp := new(struct {
+		ID      string   `json:"id"`
+		JsonRPC string   `json:"jsonrpc"`
+		Result  []string `json:"result"`
+	})
+
+	if err := requestJSON(ctx, g.httpClient, http.MethodGet, "/", req, &resp); err != nil {
+		return nil, err
+	}
+
+	return resp.Result, nil
+}
+
+// contains checks if list contains string
+func contains(list []string, find string) bool {
+	for _, v := range list {
+		if v == find {
+			return true
+		}
+	}
+
+	return false
+}
+
+// addPrefix adds prefix to string if it doesn't exist
+func addPrefix(prefix, to string) string {
+	if !strings.HasPrefix(to, prefix) {
+		return prefix + to
+	}
+	return to
+}
+
+// removePrefix removes prefix from string if it exists
+func removePrefix(prefix, from string) string {
+	if strings.HasPrefix(from, prefix) {
+		return from[len(prefix):]
+	}
+	return from
+}
+
+// float64ToBigInt converts float64 to big.Int
+func float64ToBigInt(f float64) *big.Int {
```

Comment:
you seem to be using that both for bzz and eth even though they have different decimals